File: SpringBoot/src/main/java/com/project/foret/entity/Board.java
package com.project.foret.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Board {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int type;
    private int hit;
    private String subject;
    private String content;
    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
    private Date reg_date;
    private Date edit_date;

    @ManyToOne
    @JoinColumn(name = "writer_id")
    private Member member;
    @ManyToOne
    @JoinColumn(name = "foret_id")
    private Foret foret;
}

File: Android/app/src/main/java/com/project/foret/model/Board.kt
package com.project.foret.model

data class Board(
    val id: Long?,
    val type: Int?,
    val hit: Int?,
    val subject: String?,
    val content: String?,
    val reg_date: String?,
    val edit_date: String?,
    val member: Member?,
    val foret: Foret?,
    val photos: MutableList<Photo>?,
    val comments: MutableList<Comment>?,
    val likes: MutableList<Member>?,
    val comment_count: Int?,
    val like_count: Int?
) {
    constructor(id: Long)
            : this(id, null, null, null, null, null, null, null, null, null, null, null, null, null)
    constructor(type: Int, subject: String, content: String, member: Member, foret: Foret?)
            : this(null,type, null, subject, content, null, null, member, foret, null,  null,null, null, null)
}
